src/util/irep_serialization.h

```diff
@@ -81,32 +80,123 @@ class irep_serializationt
     }
   };
 
-  irep_serializationt(ireps_containert &ic) : ireps_container(ic)
-  {
-    read_buffer.resize(1, 0);
-    clear();
-  };
-
-  void reference_convert(std::istream &, irept &irep);
-  void reference_convert(const irept &irep, std::ostream &);
-
-  irep_idt read_string_ref(std::istream &);
-  void write_string_ref(std::ostream &, const dstring &);
-
+  /**
+   * Constructor for serialization
+   * @param ic container which will hold the serialization.
+   */
+  explicit irep_serializationt(ireps_containert &ic) : ireps_container(ic){};
+
+  /**
+   * Checks if a reference was already read and then stores it in an irept
+   * reference
+   * @param in input containing the irep
+   * @param irep object where the irep will be stored
+   */
+  void reference_convert(std::istream &in, irept &irep);
+
+  /**
+ * Checks if a reference was already written if yes, then stores it's id
```

Comment:
"Checks if a reference was already written if yes, then stores it's id if not, then stores it's id alongside with irep" -> "Checks whether a reference was already written; if so, then stores its id; otherwise, stores its id alongside with irep"